### PR TITLE
Allow providing list of services not to start

### PR DESCRIPTION
**Pull request by kobergj**
Signed-off-by: jkoberg <jkoberg@owncloud.com>

#### Allow providing list of services NOT to start

Until now if one wanted to use a custom version of a service, one
needed to provide `OCIS_RUN_SERVICES` which is a list of all services to start.
Now one can provide `OCIS_EXCLUDE_RUN_SERVICES` which is a list of only services not to start

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: ocis-pkg/config/config.go [hunk 1]
```diff
@@ -49,7 +49,8 @@ type Mode int
 type Runtime struct {
 	Port       string `yaml:"port" env:"OCIS_RUNTIME_PORT"`
 	Host       string `yaml:"host" env:"OCIS_RUNTIME_HOST"`
-	Extensions string `yaml:"services" env:"OCIS_RUN_EXTENSIONS;OCIS_RUN_SERVICES"`
+	Extensions string `yaml:"services" env:"OCIS_RUN_EXTENSIONS;OCIS_RUN_SERVICES" desc:"Expects a space seperated list of service names. Will start only the listed services."`
+	Disabled   string `yaml:"disabled_services" env:"OCIS_EXCLUDE_RUN_SERVICES" desc:"Expects a space seperated list of service names. Will start all services except of the ones listed. Has no effect when OCIS_RUN_SERVICES is set."`
```
Comment (phil-davis) [inline ocis-pkg/config/config.go:53]:
```suggestion
	Extensions string `yaml:"services" env:"OCIS_RUN_EXTENSIONS;OCIS_RUN_SERVICES" desc:"Expects a space separated list of service names. Will start only the listed services."`
	Disabled   string `yaml:"disabled_services" env:"OCIS_EXCLUDE_RUN_SERVICES" desc:"Expects a space separated list of service names. Will start all services except for the ones listed. Has no effect when OCIS_RUN_SERVICES is set."`
```

File: ocis/pkg/runtime/service/service.go [hunk 2]
```diff
@@ -248,12 +248,24 @@ func (s *Service) generateRunSet(cfg *ociscfg.Config) {
 		return
 	}
 
+	disabled := make(map[string]bool)
+	if cfg.Runtime.Disabled != "" {
+		e := strings.Split(strings.ReplaceAll(cfg.Runtime.Disabled, " ", ""), ",")
+		for _, s := range e {
+			disabled[s] = true
```
Comment (dragonchaser) [inline ocis/pkg/runtime/service/service.go:255]: Why do the extra step and store it to a separate array, can't you just remove it from the `runset` above?

File: ocis-pkg/config/config.go (hunk 1 above)
Comment (kobergj) [reply ocis-pkg/config/config.go]: Thanks!

**Review by dragonchaser — CHANGES_REQUESTED** [review]

**Review by mmattel — APPROVED** [review]
Looks good from the description pov. Will update the docs when merged.

File: ocis/pkg/runtime/service/service.go (hunk 2 above)
Comment (kobergj) [reply ocis/pkg/runtime/service/service.go:255]: `runset` is a `[]string`. To delete from it I would need to loop over it to find the index, then delete the index from the slice. There are some other tricks how to delete from slices but all of them feel hacky to me. I think it is much cleaner with a helping map.

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-8</strong>](https://drone.owncloud.com/owncloud/ocis/13731/50/1) failed. Further test are cancelled...

File: ocis/pkg/runtime/service/service.go (hunk 2 above)
Comment (kobergj) [reply ocis/pkg/runtime/service/service.go]: Yes that works. Good idea. :+1: Now the name also makes sense because it is really a `set`

**Review by dragonchaser — APPROVED** [review]
lgtm

**Pull request merged**